Promise_NonGeneric.cs

```diff
@@ -1001,10 +1001,37 @@ public static IPromise Sequence(params Func<IPromise>[] fns)
         /// </summary>
         public static IPromise Sequence(IEnumerable<Func<IPromise>> fns)
         {
-            return fns.Aggregate(
+            var promise = new Promise();
+
+            int count = 0;
+
+            fns.Aggregate(
                 Resolved(),
-                (prevPromise, fn) => prevPromise.Then(fn)
-            );
+                (prevPromise, fn) =>
+                {
+                    int itemSequence = count;
+                    ++count;
+
+                    return prevPromise
+                            .Then(() =>
+                            {
+                                var sliceLength = 1f / count;
+                                promise.ReportProgress(sliceLength * itemSequence);
+                                return fn();
+                            })
+                            .Progress(v =>
+                            {
+                                var sliceLength = 1f / count;
+                                promise.ReportProgress(sliceLength * (v + itemSequence));
+                            })
+                    ;
+                }
+            )
+            .Then(() => promise.Resolve())
+            .Catch(promise.Reject)
+            .Done();
```

Comment:
We should not be calling Done here because it terminates the promise chain. This would break any code that calls `Sequence(...).Then(...)`

Reply:
You are right, my bad!